dashboards-observability/public/components/visualizations/charts/financial/gauge/gauge.tsx

```diff
@@ -56,17 +67,15 @@ export const Gauge = ({ visualizations, layout, config }: any) => {
       // case 3: multiple dimensions and multiple metrics
       if (dimensionsLength && metricsLength) {
         const selectedDimensionsData = [
-          ...dimensions.map((dimension: any) =>
-            data[dimension.name].slice(0, DisplayDefaultGauges)
-          ),
+          ...dimensions.map((dimension: any) => data[dimension.name].slice(0, numberOfGauges)),
         ].reduce(function (prev, cur) {
           return prev.map(function (i, j) {
             return `${i}, ${cur[j]}`;
           });
         });
 
         const selectedMetricsData = [
-          ...metrics.map((metric: any) => data[metric.name].slice(0, DisplayDefaultGauges)),
+          ...metrics.map((metric: any) => data[metric.name].slice(0, numberOfGauges)),
```

Comment:
You do not need this additional destructuring here. map already returns a new array.

Reply:
selectedMetricsData: array of array: it is used to get separate list of only selected metrics data from "data query response object". 

i tried to iterate on array of array as we have multiple metrics and multiple dimensions.  values are picked from inner array based on index of that inner array for both metrics and dimensions

same is case for dimensions. created separate array of concatenated labels using array of selected dimensions

@spattnaik